qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/MapboxNavigationViewOfflineOnlineRouteSwitchActivity.kt

```diff
@@ -0,0 +1,190 @@
+import com.mapbox.navigation.qa_test_app.utils.startActivity
+import com.mapbox.navigation.qa_test_app.view.base.DrawerActivity
+import com.mapbox.navigation.qa_test_app.view.customnavview.NavigationViewController
+import com.mapbox.navigation.utils.internal.toPoint
+import com.mapbox.turf.TurfMeasurement
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.flatMapMerge
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalMapboxNavigationAPI::class)
+class MapboxNavigationViewOfflineOnlineRouteSwitchActivity : DrawerActivity() {
+
+    private lateinit var binding: LayoutActivityNavigationOfflineOnlineViewBinding
+    private lateinit var menuBinding: LayoutDrawerMenuNavViewBinding
+
+    override fun onCreateContentView(): View {
+        binding = LayoutActivityNavigationOfflineOnlineViewBinding.inflate(layoutInflater)
+        return binding.root
+    }
+
+    override fun onCreateMenuView(): View {
+        menuBinding = LayoutDrawerMenuNavViewBinding.inflate(layoutInflater)
+        return menuBinding.root
+    }
+
+    private lateinit var controller: NavigationViewController
+
+    private var onlineRouteSwitch = OnlineRouteAlternativesSwitch()
+
+    private val tilesLoadingObserver = LoadAreaAroundObserver { state ->
+        setTitle("${state.requestedTiles}/${state.requestedTiles}, ${state.primaryRouteId}")
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        val tileStore = TileStore.create()
+        tileStore.setOption(
+            TileStoreOptions.MAPBOX_ACCESS_TOKEN,
+            TileDataDomain.NAVIGATION,
+            Value.valueOf(getString(R.string.mapbox_access_token))
+        )
+        MapboxNavigationApp.setup(
+            NavigationOptions.Builder(applicationContext)
+                .accessToken(getString(R.string.mapbox_access_token))
+                .routingTilesOptions(
+                    RoutingTilesOptions.Builder()
+                        .tileStore(tileStore)
+                        .build()
+                )
+                .build()
+        )
+        super.onCreate(savedInstanceState)
+        controller = NavigationViewController(this, binding.navigationView)
+
+        menuBinding.toggleReplay.isChecked = binding.navigationView.api.isReplayEnabled()
+        menuBinding.toggleReplay.setOnCheckedChangeListener { _, isChecked ->
+            binding.navigationView.api.routeReplayEnabled(isChecked)
+        }
+
+        binding.navigationView.setRouteOptionsInterceptor {
+            it.profile(DirectionsCriteria.PROFILE_WALKING)
```

Comment:
Why?

Reply:
I tested it manually walking in the park near my home. Didn't want to drive and test simultaneously 🙂

Reply:
What about route replay? We can use it to test and we'll probably be using it and not walk. :)
I mean that the case with switching to online is especially important for EV and traffic. Walking does not really need it.